tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/InstrumentationDefinitionsGenerator.cs

```diff
@@ -0,0 +1,771 @@
+                }
+
+                if (typeNames is null or { Length: 0 } && typeName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.TypeName,
+                            Constants.InstrumentAttributeProperties.TypeNames));
+                }
+
+                if (integrationName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.IntegrationName));
+                }
+
+                if (methodName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.MethodName));
+                }
+
+                if (returnTypeName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.MethodName));
+                }
+
+                (ushort Major, ushort Minor, ushort Patch) minVersion = default;
+                (ushort Major, ushort Minor, ushort Patch) maxVersion = default;
+                if (minimumVersion is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.MinimumVersion));
+                }
+                else if (!TryGetVersion(minimumVersion, ushort.MinValue, out minVersion))
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        InvalidVersionFormatDiagnostic.Create(
```

Comment:
With the current InvalidVersionFormatDiagnostic, is the warning message specific enough to identify that it's the MinimumVersion that is invalid?

Reply:
Good catch - updated the message to call out the property at fault in https://github.com/DataDog/dd-trace-dotnet/pull/2288/commits/bb06fe0e6040cc2707f475d33e03a3e300719ddf